### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by wi1dcard**

### CODE REVIEW

**Comment by clearloop** [comment]
Have you every yarn before editing files? @wi1dcard 

The conventional commit message seems not triggered.

**Comment by wi1dcard** [comment]
> Have you every `yarn` before editing files? @wi1dcard
> 
> The conventional commit message seems not triggered.

Nope. You mean we set up the Git hook checking the commit message format in order to make sure it conforms to Conventional Commits?

Although personally I prefer the commit message just to be 1) written in English, 2) written in natural language (instead of code language), 3) describes the propose of the commit. I still look forward to having a commit message style guide for us developers in every single repo. As you said before, I believe the Conventional Commits is a good example.

Since all the repos are not maintained by a single developer, I strongly suggest we have a discussion and make a decision which commits style we should use. Then setup on CI instead of using Git hooks. This will make a lot of advantages, for example:

1. We can not require all the developers (just like me) to run `yarn install` before they contribute. Some of the developers prefer using Web IDE, or perhaps they just want to correct a small typo in docs.
2. Git hooks make `git commit` running longer than without hooks.
3. While we are developing in feature branches, there is a chance that we just commit for testing and at that time we have no idea of the accurate commit message. Since CI failures can block PRs, however, Git hooks block commits. We have to make sure every commit conforms to lint.
4. Finally, the most important issue I would have to say, I don't really like there's any "hidden mechanism" modifying my git config, affecting git operations. For people like me who have no idea of the config, I have to read husky docs to get understand why git is not performing as always. That might be bothering.

Hope we can improve the style guide of commits and its lint process!

**Comment by wi1dcard** [comment]
@clearloop Fixed commit messages. The PR is ready for review.

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -45,16 +61,27 @@ jobs:
...
+          path: ${{ steps.yarn_cache_dir.outputs.dir }}
+          key: ${{ matrix.os }}-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ matrix.os }}-${{ matrix.node-version }}-yarn-
+
+      - run: yarn && yarn deps
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: lib-${{ matrix.os }}
```
Comment (clearloop) [inline .github/workflows/nodejs.yml:85]: This is actually what this repo need!

Comment (wi1dcard) [reply .github/workflows/nodejs.yml:85]: > This is actually what this repo needs!

@clearloop lol, there's some great explanation of the concept "artifacts" - thought that might be helpful for you comprehend it:

<https://docs.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts#about-workflow-artifacts>

**Pull request merged**